src/js_native_api_v8.cc

```diff
@@ -866,26 +866,67 @@ napi_status napi_define_class(napi_env env,
 napi_status napi_get_property_names(napi_env env,
                                     napi_value object,
                                     napi_value* result) {
+  return napi_get_all_property_names(
+      env,
+      object,
+      napi_key_include_all_prototypes,
+      static_cast<napi_key_filter>(napi_key_enumerable |
+                                   napi_key_skip_symbols),
+      napi_key_numbers_to_strings,
+      result);
+}
+
+napi_status napi_get_all_property_names(napi_env env,
+                                        napi_value object,
+                                        napi_key_collection_mode key_mode,
+                                        napi_key_filter key_filter,
+                                        napi_key_conversion key_conversion,
+                                        napi_value* result) {
   NAPI_PREAMBLE(env);
   CHECK_ARG(env, result);
 
   v8::Local<v8::Context> context = env->context();
   v8::Local<v8::Object> obj;
   CHECK_TO_OBJECT(env, context, obj, object);
 
-  v8::MaybeLocal<v8::Array> maybe_propertynames = obj->GetPropertyNames(
-    context,
-    v8::KeyCollectionMode::kIncludePrototypes,
-    static_cast<v8::PropertyFilter>(
-        v8::PropertyFilter::ONLY_ENUMERABLE |
-        v8::PropertyFilter::SKIP_SYMBOLS),
-    v8::IndexFilter::kIncludeIndices,
-    v8::KeyConversionMode::kConvertToString);
+  v8::PropertyFilter filter = static_cast<v8::PropertyFilter>(key_filter);
```

Comment:
Can we have a `switch` for these values as we do for the values below?

Reply:
how to handle like this using `swtich`?
```cpo
static_cast<napi_key_filter>(napi_key_enumerable | napi_key_skip_symbols)
```

Reply:
How about a series of if-statements then:
```suggestion
  v8::PropertyFilter filter = v8::PropertyFilter::ALL_PROPERTIES;
  if (key_filter | napi_key_writable) {
    filter != v8::PropertyFilter::ONLY_WRITABLE;
  }
  if (key_filter | napi_key_enumerable) {
    filter != v8::PropertyFilter::ONLY_ENUMERABLE;
  }
  // etc.
```